asyncio_run_in_process/process.py

```diff
@@ -0,0 +1,195 @@
+import asyncio
+import os
+import signal
+from typing import Callable, Optional, Sequence
+
+from ._utils import pickle_value
+from .abc import ProcessAPI
+from .exceptions import ProcessKilled
+from .state import State
+from .typing import TReturn
+
+
+class Process(ProcessAPI[TReturn]):
+    _pid: Optional[int] = None
+    _returncode: Optional[int] = None
+    _return_value: TReturn
+    _error: Optional[BaseException] = None
+    _state: State = State.INITIALIZING
+
+    sub_proc_payload: bytes
+
+    def __init__(
+        self, async_fn: Callable[..., TReturn], args: Sequence[TReturn]
+    ) -> None:
+        self._async_fn = async_fn
+        self._args = args
+        self.sub_proc_payload = pickle_value((self._async_fn, self._args))
+
+        self._has_pid = asyncio.Event()
+        self._has_returncode = asyncio.Event()
+        self._has_return_value = asyncio.Event()
+        self._has_error = asyncio.Event()
+        self._state_changed = asyncio.Event()
+
+    #
+    # State
+    #
+    @property
+    def state(self) -> State:
+        return self._state
+
+    @state.setter
+    def state(self, value: State) -> None:
+        self._state = value
+        self._state_changed.set()
+        self._state_changed = asyncio.Event()
+
+    async def wait_for_state(self, state: State) -> None:
+        """
+        Block until the process as reached the
+        """
```

Comment:
I suspect you passed out during the middle of writing the docstring :point_up:  :sweat_smile: